pallets/did/src/lib.rs

```diff
@@ -1140,9 +1152,139 @@ pub mod pallet {
+		#[pallet::weight({
+			let dispatch_info = call.get_dispatch_info();
+
+			(<T as pallet::Config>::WeightInfo::dispatch_as().saturating_add(dispatch_info.weight), dispatch_info.class)
+		})]
+		pub fn dispatch_as(
+			origin: OriginFor<T>,
+			did_identifier: DidIdentifierOf<T>,
+			call: Box<DidCallableOf<T>>,
+		) -> DispatchResultWithPostInfo {
+			let who = ensure_signed(origin)?;
+
+			let verification_key_relationship =
+				call.derive_verification_key_relationship().map_err(Error::<T>::from)?;
+
+			Pallet::<T>::verify_account_authorization(&did_identifier, &who, verification_key_relationship)
+				.map_err(Error::<T>::from)?;
+
+			log::debug!("Dispatch call from DID {:?}", did_identifier);
+
+			#[cfg(not(feature = "runtime-benchmarks"))]
+			let result = call.dispatch(
+				DidRawOrigin {
+					id: did_identifier.clone(),
+					submitter: who,
+				}
+				.into(),
+			);
+			#[cfg(feature = "runtime-benchmarks")]
+			let result = call.dispatch(RawOrigin::Signed(did_identifier.clone()).into());
+
+			let dispatch_event_payload = result.map(|_| ()).map_err(|e| e.error);
+
+			Self::deposit_event(Event::DidCallDispatched(did_identifier, dispatch_event_payload));
+
+			result
+		}
+
+		/// Store a new DID on chain.
+		///
+		/// The DID identifier is derived from the account ID that submits this
+		/// call. The authentication key must correspond to the account ID that
+		/// submitted this call. For accounts that use the ed25519 and sr25519
+		/// schema, the authentication key must be of the
+		/// `DidVerificationKey::Ed25519` or `DidVerificationKey::Sr25519`
+		/// variant and contains the public key. For Ecdsa accounts, the
+		/// `DidVerificationKey::Ecdsa` variant is calculated by hashing the
+		/// Ecdsa public key.
+		///
+		/// If this call is dispatched by an account id that doesn't correspond
+		/// to a public private key pair, the `DidVerificationKey::Account`
+		/// variant shall be used (Multisig, Pure Proxy, Governance origins).
+		/// The resulting DID can NOT be used for signing data and is therefore
+		/// limited to onchain activities.
+		///
+		/// There must be no DID information stored on chain under the same DID
+		/// identifier. This call will fail if there exists a DID with the same
+		/// identifier or if a DID with the same identifier existed and was
+		/// deleted.
+		///
+		/// The origin for this account must be funded and provide the required
+		/// deposit and fee.
+		///
+		/// Emits `DidCreated`.
+		#[pallet::call_index(16)]
+		#[pallet::weight(<T as pallet::Config>::WeightInfo::create_from_account())]
+		pub fn create_from_account(
+			origin: OriginFor<T>,
+			authentication_key: DidVerificationKey<AccountIdOf<T>>,
+		) -> DispatchResult {
+			let sender = ensure_signed(origin)?;
+			let did_identifier: DidIdentifierOf<T> = sender.clone().into();
+			let auth_key_id: AccountIdOf<T> = authentication_key.into_account();
+			ensure!(auth_key_id == sender, Error::<T>::BadDidOrigin);
```

Comment:
Why this line here? So if you are using one account, you can't use a key different than the one that matches the origin's account? I thought this additional extrinsic was only supposed to remove the signature requirement.

Reply:
The origin has to be the authentication key. This is basically the check that the owner of the authentication key agrees to creating the DID.
This basically replaces the second DID signature that is required for the old `create` call.
If we would not have this check, I could create DIDs for any public key I want.

Reply:
Ah ok. Now I understand. That would be no different than just changing the authentication key immediately after creating the DID.

Reply:
[a8090e5](https://github.com/KILTprotocol/kilt-node/pull/551/commits/a8090e5243bc73e6eb55f9d2e1d3fef92e1985e9)